webapp/src/js/directives/header.js

```diff
@@ -50,6 +53,41 @@ angular.module('inboxDirectives').directive('mmHeader', function() {
         DBSync.sync(true);
       };
 
+      ctrl.isMobile = () => {
+        return responsive.isMobile();
+      }; 
+
+      const tabs = [{name:'messages',state:'messages.detail',icon:'fa-envelope',translation:'Messages',
+        permissions:[['can_view_messages'],['can_view_messages_tab']]},
+      {name:'tasks',state:'tasks.detail',icon:'fa-flag',translation:'Tasks',
+        permissions:[['can_view_tasks'],['can_view_tasks_tab']]},
+      {name:'reports',state:'reports.detail',icon:'fa-list-alt',translation:'Reports',
+        permissions:[['can_view_reports'],['can_view_reports_tab']]},
+      {name:'contacts',state:'contacts.detail',icon:'fa-user',translation:'Contacts',
+        permissions:[['can_view_contacts'],['can_view_contacts_tab']]},
+      {name:'analytics',state:'analytics',icon:'fa-bar-chart-o',translation:'Analytics',
+        permissions:[['can_view_analytics'],['can_view_analytics_tab']]}];
+
+      const permitedTabs = () => {
+        const result = [];
+        tabs.forEach(tab => {
+          Auth.any(tab.permissions)
+            .then(() =>  {
+              result.push(tab);
+            })
+            .catch(err => {
+              $log.error('Error determining tab permissions', err);
+            });
+        });
+        return result;
+      };
+
+      const result = permitedTabs();
```

Comment:
A good tip is that the descriptiveness of your variable name should be commensurate to the size of the scope it exists in.

This means that it's OK to use a vague variable name when you have a one line function: `const foos = bars.map(b => b.foo)`, but it probably wouldn't be OK if you had:

```js
function someApiFunction(b) {
  // many lines of code
}
```

With this in mind, reconsider the use of the very generic "result" here.